app/src/main/java/com/android/solvit/shared/model/provider/ProviderRepositoryFirestore.kt

```diff
@@ -39,6 +40,38 @@ class ProviderRepositoryFirestore(
       val languages = (doc.get("languages") as List<*>).map { Language.valueOf(it as String) }
       val companyName = doc.getString("companyName") ?: ""
       val phone = doc.getString("phone") ?: ""
+
+      // Convert schedule
+      val scheduleDoc = doc.get("schedule") as? Map<*, *>
+      val schedule =
```

Comment:
Is it possible to extract this conversion code and put it in a function (just to ease readability)?
The rest looks good!

Reply:
You're right it would greatly improve readability. I have refactored it and added KDoc documentation for the schedule-related functions